crates/ika-core/src/dwallet_mpc/mpc_session.rs

```diff
@@ -118,65 +151,32 @@ impl DWalletMPCSession {
             session_identifier=?message.session_identifier,
             from_authority=?message.authority,
             receiving_authority=?self.validator_name,
-            mpc_round=?mpc_round_number,
+            consensus_round=?consensus_round,
             message_size_bytes=?message.message.len(),
             ?mpc_protocol,
             "Received a dWallet MPC message",
         );
 
-        if sender_party_id == self.party_id && self.current_mpc_round <= mpc_round_number {
-            // Received a message from ourselves from the consensus, so it's safe to advance the round.
-            let new_mpc_round = mpc_round_number + 1;
-            info!(
-                session_identifier=?message.session_identifier,
-                authority=?self.validator_name,
-                message_mpc_round=?mpc_round_number,
-                current_mpc_round=self.current_mpc_round,
-                new_mpc_round,
-                mpc_protocol,
-                "Advancing current MPC round",
-            );
-
-            self.current_mpc_round = new_mpc_round;
-        }
-
-        let consensus_round_messages_map = self
-            .messages_by_consensus_round
-            .entry(consensus_round)
-            .or_default();
-
-        if let Vacant(e) = consensus_round_messages_map.entry(sender_party_id) {
-            e.insert(message.message);
-        }
-    }
-
-    /// Records a threshold not reached error that we got when advancing
-    /// this session with messages up to `consensus_round`.
-    pub(crate) fn record_threshold_not_reached(&mut self, consensus_round: u64) {
-        let MPCSessionStatus::Active { request, .. } = &self.status else {
+        let ComputationType::MPC {
+            messages_by_consensus_round,
+        } = &mut self.computation_type
+        else {
             error!(
                 should_never_happen=true,
                 session_identifier=?self.session_identifier,
-                "tried to record threshold not reached for a non-active MPC session"
+                computation_type=?self.computation_type,
+                "tried to add a message to a non-MPC session"
             );
             return;
```

Comment:
You just moved the bug here, a malicious validator can send a message for a native session, you won't report it as malicious, and you have `should_never_happen=true,` which is wrong